### ISSUE TITLE
Excluding functions ignoring symlinks

### ISSUE THREAD

**Issue by jakunow** [ncw/rclone#2671]
Hi,

Using the following:
rclone v1.44
- os/arch: linux/amd64
- go version: go1.11.1

I know there were some changes merged that involved symlinks (like #1509 and #2312 ). However I'm still experiencing behavior that I believe is incorrect. Rclone does not seem to exclude a file if that file is a symlink which still ends with an error being logged.

In #2312 I've mentioned that

> Seems that symlinks are evaluated before processing filtering.

Is this expected behavior?

**Comment by ncw** [comment]
Yes symlinks are processed before filters which is a little unfortunate, but the way the rclone architecture works.

What problem is this causing you?

**Comment by jakunow** [comment]
Problem I'm seeing is with logging and not rclone logic itself. 

Broken symlinks will create ERROR entries in log file whenever rclone runs. It's hard to monitor any really issues with logfile. Perhaps it would be possible to disable logging for those events instead of changing whole architecture?

**Comment by 0x3333** [comment]
What needs to be done to avoid these errors? I may step in to fix this.

**Comment by roucc** [comment]
Sorry, this is actually fixed after testing it. Here is my test:

```console
~ ❯ mkdir /tmp/lkjffdslkjfs

~ ❯ cd /tmp/lkjffdslkjfs/

/tmp/lkjffdslkjfs ❯ ln -s bananapeanut file.txt

/tmp/lkjffdslkjfs ❯ ll
total 0
lrwxrwxrwx 1 dougal dougal 12 Nov 27 14:13 file.txt -> bananapeanut

/tmp/lkjffdslkjfs ❯ cat file.txt 
cat: file.txt: No such file or directory

/tmp/lkjffdslkjfs ✗ rclone lsf .
2025/11/27 14:14:19 NOTICE: file.txt: Can't follow symlink without -L/--copy-links

/tmp/lkjffdslkjfs ❯ rclone lsf . -L
2025/11/27 14:14:25 ERROR : file.txt: Listing error: symlink: stat /tmp/lkjffdslkjfs/file.txt: no such file or directory
2025/11/27 14:14:25 NOTICE: Failed to lsf: symlink: stat /tmp/lkjffdslkjfs/file.txt: no such file or directory

/tmp/lkjffdslkjfs ✗ rclone lsf . -L --exclude file.txt 

/tmp/lkjffdslkjfs ❯ 
```

Here you can see excluding the symlink did not produce an error!
